testing/tests/DevExpress.ui.widgets.dataGrid/keyboardNavigation.tests.js

```diff
@@ -4877,6 +4877,32 @@ QUnit.module("Keyboard navigation with real dataController and columnsController
         assert.notOk($cell.hasClass("dx-focused"), "cell has .dx-focused");
     });
 
+    QUnit.testInActiveWindow("Cell should not have focus after click if allowediting=false and edit mode == 'cell || batch' (T657612)", function(assert) {
```

Comment:
allowediting -> allowEditing, Also conditions are very strange and inconsistent.

Reply:
Corrected